### PR TITLE
cleanup & fix set channel positions

### PR DESCRIPTION
**Pull request by appellation**
**Please describe the changes this PR makes and why it should be merged:**
- Makes things more OOP
- Removes code duplication
- Fixes `Util#moveElementInArray` to allow positions outside the bounds of the array

**Semantic versioning classification:**  
- [ ] This PR changes the library's interface (methods or parameters added)
  - [x] This PR includes breaking changes (methods removed or renamed, parameters moved or removed)
- [ ] This PR **only** includes non-code changes, like changes to documentation, README, etc.

### CODE REVIEW

File: src/structures/GuildChannel.js [hunk 1]
```diff
@@ -275,8 +276,14 @@ class GuildChannel extends Channel {
    *   .then(newChannel => console.log(`Channel's new position is ${newChannel.position}`))
    *   .catch(console.error);
    */
-  setPosition(position, relative) {
-    return this.guild.setChannelPosition(this, position, relative).then(() => this);
+  setPosition(position, relative = false) {
+    position = Number(position);
+    if (isNaN(position)) return Promise.reject(new TypeError('INVALID_TYPE', 'position', 'number'));
+
+    let updatedChannels = this.guild._sortedChannels(this.type).array();
```
Comment (devsnek) [inline src/structures/GuildChannel.js]: the call for sorted channels takes a channel not a channel type

**Comment by PgBiel** [comment]
Why have reasons for position changes when they don't appear in audit logs?

**Comment by iCrawl** [comment]
You could say the same about a bot deleting your messages.

**Comment by iCrawl** [comment]
Properly handled in #1919, where it covers all cases.

**Pull request closed without merging**